spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -93,7 +94,8 @@
 
 	private final RedisTemplate<String, RedisLock> redisTemplate;
 
-	private final ThreadLocal<List<RedisLock>> threadLocks = new ThreadLocal<List<RedisLock>>();
+	private final ThreadLocal<Set<RedisLock>> weakThreadLocks = new ThreadLocal<Set<RedisLock>>();
+	private final ThreadLocal<List<RedisLock>> hardThreadLocks = new ThreadLocal<List<RedisLock>>();
```

Comment:
And one more thing I forgot, your code is not thread safe coz you get rid of ThreadLocal so you would need additional synchronization which I'd rather avoid. For example - we use this Registry under pretty high load so additional synchronization may cause lots of locking in our multithreaded application (as it does when we used default local LockRegistry inside this RedisLockRegistry), so any additional sync may cost alot.